src/Repository/EmailVerificationTokenRepository.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace BigGive\Identity\Repository;
+
+use BigGive\Identity\Domain\EmailVerificationToken;
+use Doctrine\ORM\EntityManagerInterface;
+
+class EmailVerificationTokenRepository
+{
+    /** @psalm-suppress PossiblyUnusedMethod - used in DI */
+    public function __construct(private EntityManagerInterface $em)
+    {
+    }
+
+    public function findToken(
```

Comment:
this is suspiciously similar to https://github.com/thebiggive/matchbot/blob/ab0ba2e04a9f901006332f4e3a3268e6c033a2bd/src/Domain/EmailVerificationTokenRepository.php#L17 - although in that one we don't use the secret as part of the query.